packages/cohere/src/cohere-chat-language-model.test.ts

```diff
@@ -278,6 +475,442 @@ describe('doStream', () => {
+      },
+    });
+
+    const responseArray = await convertReadableStreamToArray(stream);
+
+    expect(responseArray).toStrictEqual([
+      {
+        type: 'tool-call-delta',
+        toolCallType: 'function',
+        toolCallId: expect.any(String),
+        toolName: 'test-tool',
+        argsTextDelta: '',
+      },
+      {
+        type: 'tool-call-delta',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        argsTextDelta: '{\n    "',
+      },
+      {
+        type: 'tool-call-delta',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        argsTextDelta: 'ticker',
+      },
+      {
+        type: 'tool-call-delta',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        argsTextDelta: '_',
+      },
+      {
+        type: 'tool-call-delta',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        argsTextDelta: 'symbol',
+      },
+      {
+        type: 'tool-call-delta',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        argsTextDelta: '":',
+      },
+      {
+        type: 'tool-call-delta',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        argsTextDelta: ' "',
+      },
+      {
+        type: 'tool-call-delta',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        argsTextDelta: 'AAPL',
+      },
+      {
+        type: 'tool-call-delta',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        argsTextDelta: '"',
+      },
+      {
+        type: 'tool-call-delta',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        argsTextDelta: '\n',
+      },
+      {
+        type: 'tool-call-delta',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        argsTextDelta: '}',
+      },
+      {
+        type: 'tool-call',
+        toolCallId: expect.any(String),
+        toolCallType: 'function',
+        toolName: 'test-tool',
+        args: '{"ticker_symbol":"AAPL"}',
+      },
+      {
+        finishReason: 'stop',
+        type: 'finish',
+        usage: {
+          completionTokens: 62,
+          promptTokens: 893,
+        },
+      },
+    ]);
+
+    // Check if the tool call ID is the same in the tool call delta and the tool call
+
+    const toolCallDelta = responseArray.find(
+      element => element.type === 'tool-call-delta',
+    );
+
+    const toolCall = responseArray.find(
+      element => element.type === 'tool-call',
+    );
+
+    expect(toolCallDelta?.toolCallId).toBe(toolCall?.toolCallId);
```

Comment:
this testing involving `generateId` is a little bit brittle. one way to improve it is to inject a `generateId` function in tests, e.g. as in the vertex test or earlier versions of the mistral and anthropic tests (before they had tool ids): https://github.com/vercel/ai/blob/main/packages/google-vertex/src/google-vertex-language-model.test.ts#L28

Reply:
til, thanks! updated the tests to use custom `generateId`